entities/autocut/autocut.go

```diff
@@ -0,0 +1,47 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package autocut
+
+func Autocut(yValues []float32, cutOff int) int {
+	diff := make([]float32, len(yValues))
+	step := 1. / (float32(len(yValues)) - 1.)
```

Comment:
potential div/0

Reply:
for floats this results in +inf 😅 

added an early return